python/paddle/fluid/executor.py

```diff
@@ -404,29 +404,30 @@ def _as_lodtensor(data, place, dtype=None):
             >>>     ...
 
         Args:
-            data(numpy.ndarray): a instance of array
+            data(numpy.ndarray): a instance of array, scaler, list or tuple
             data(core.Place): the place of created tensor
             dtype(core.VarDesc.VarType): the expected data type of created tensor
 
         Returns:
             LoDTensor
         """
-    if isinstance(data, list):
-        raise RuntimeError("Some of your feed data hold LoD information. \
-                They can not be completely cast from a list of Python \
-                ndarray to LoDTensor. Please convert data to LoDTensor \
-                directly before feeding the data.\
-                ")
-
-    #NOTE(zhiqiu): convert python builtin ,like float and int, to numpy array
+    #NOTE(zhiqiu): convert python builtin, like float, int, and list, to numpy ndarray
     if not isinstance(data, np.ndarray):
+        assert dtype is not None, 'The dtype should be given when feed data is not np.ndarray'
         if np.isscalar(data):
-            assert dtype is not None, 'dtype should be given when casting python scalar to tensor'
             dtype = convert_dtype(dtype) if isinstance(
                 dtype, core.VarDesc.VarType) else dtype
             data = np.array([data]).astype(dtype)
+        elif isinstance(data, (list, tuple)):
+            dtype = convert_dtype(dtype) if isinstance(
+                dtype, core.VarDesc.VarType) else dtype
+            data = np.array(data).astype(dtype)
```

Comment:
does it need to add check here like https://github.com/PaddlePaddle/Paddle/blob/7b7e605189779e93173b650662cabb06c2d697d7/python/paddle/fluid/reader.py#L811-L819

Reply:
Yes, thx.